cmake/SSGCommon.cmake

```diff
@@ -153,57 +153,50 @@ macro(ssg_build_ocil_unlinked PRODUCT)
+        COMMAND "${CMAKE_COMMAND}" -E remove_directory "${BUILD_REMEDIATIONS_DIR}"
+        COMMAND env "PYTHONPATH=$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE}" "${SSG_BUILD_SCRIPTS}/generate_from_templates.py" --languages ${LANGUAGES} --input "${CMAKE_CURRENT_SOURCE_DIR}/templates" "${SSG_SHARED}/templates" --output "${BUILD_REMEDIATIONS_DIR}" "${BUILD_REMEDIATIONS_DIR}/shared" --shared "${SSG_SHARED}" build
         DEPENDS generate-internal-bash-remediation-functions.xml
         DEPENDS "${CMAKE_BINARY_DIR}/bash-remediation-functions.xml"
         DEPENDS ${LANGUAGE_REMEDIATIONS_DEPENDS}
-        DEPENDS ${SHARED_LANGUAGE_REMEDIATIONS_DEPENDS}
-        DEPENDS ${EXTRA_LANGUAGE_DEPENDS}
-        DEPENDS ${EXTRA_SHARED_LANGUAGE_DEPENDS}
         DEPENDS "${SSG_BUILD_SCRIPTS}/generate_from_templates.py"
-        DEPENDS "${SSG_BUILD_SCRIPTS}/combine_remediations.py"
-        COMMENT "[${PRODUCT}-content] generating ${LANGUAGE}-fixes.xml"
-    )
-    add_custom_target(
-        generate-internal-${PRODUCT}-${LANGUAGE}-fixes.xml
-        DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/${LANGUAGE}-fixes.xml"
-    )
+        COMMENT "[${PRODUCT}-content] generating all fixes: ${LANGUAGES}"
+    )
+    foreach(LANGUAGE ${LANGUAGES})
+      file(GLOB EXTRA_LANGUAGE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/fixes/${LANGUAGE}/*")
+      file(GLOB EXTRA_SHARED_LANGUAGE_DEPENDS "${SSG_SHARED}/fixes/${LANGUAGE}/*")
+
+      add_custom_command(
+          OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${LANGUAGE}-fixes.xml"
+          COMMAND env "PYTHONPATH=$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE}" "${SSG_BUILD_SCRIPTS}/combine_remediations.py" --build-config-yaml "${CMAKE_BINARY_DIR}/build_config.yml" --product-yaml "${CMAKE_CURRENT_SOURCE_DIR}/product.yml" --remediation_type "${LANGUAGE}" --build_dir "${CMAKE_BINARY_DIR}" --output "${CMAKE_CURRENT_BINARY_DIR}/${LANGUAGE}-fixes.xml" "${BUILD_REMEDIATIONS_DIR}/shared/${LANGUAGE}" "${SSG_SHARED}/fixes/${LANGUAGE}" "${BUILD_REMEDIATIONS_DIR}/${LANGUAGE}" "${CMAKE_CURRENT_SOURCE_DIR}/fixes/${LANGUAGE}"
+          DEPENDS ${LANGUAGE_REMEDIATIONS_OUTPUTS}
+          DEPENDS ${EXTRA_LANGUAGE_DEPENDS}
+          DEPENDS ${EXTRA_SHARED_LANGUAGE_DEPENDS}
+          DEPENDS "${SSG_BUILD_SCRIPTS}/combine_remediations.py"
+          COMMENT "[${PRODUCT}-content] generating ${LANGUAGE}-fixes.xml"
+      )
+      add_custom_target(
+          generate-internal-${PRODUCT}-${LANGUAGE}-fixes.xml
+          DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/${LANGUAGE}-fixes.xml"
+      )
+    endforeach()
 
 if (SSG_SHELLCHECK_BASH_FIXES_VALIDATION_ENABLED AND SHELLCHECK_EXECUTABLE AND "${LANGUAGE}" STREQUAL "bash")
```

Comment:
this doesn't work anymore, right? there is no LANGUAGE in this macro

Reply:
maybe we can remove the shellcheck test completely because it never really did anything useful. It would be much more useful to shellcheck the final remediation roles.

Reply:
I'll remove them,  good catch. Otherwise, I could move it up as part of the foreach loop.